.github/workflows/build-test.yml

```diff
@@ -0,0 +1,169 @@
+name: build-test
+on:
+  pull_request:
+    paths-ignore:
+      - 'README.md'
+      - 'CHANGELOG.md'
+      - '.changelog/**'
+      - '.tours/**'
+      - 'contributing/**'
+      - 'demo/**'
+      - 'dev/**'
+      - 'e2e/terraform/**'
+      - 'e2e/ui/**'
+      - 'integrations/**'
+      - 'pkg/**'
+      - 'scripts/**'
+      - 'terraform/**'
+      - 'ui/**'
+      - 'website/**'
+  push:
+    branches:
+      - main
+      - release/**
+    paths-ignore:
+      - 'README.md'
+      - 'CHANGELOG.md'
+      - '.changelog/**'
+      - '.tours/**'
+      - 'contributing/**'
+      - 'demo/**'
+      - 'dev/**'
+      - 'e2e/terraform/**'
+      - 'e2e/ui/**'
+      - 'integrations/**'
+      - 'pkg/**'
+      - 'scripts/**'
+      - 'terraform/**'
+      - 'ui/**'
+      - 'website/**'
+
+env:
+  VAULT_VERSION: 1.4.1
+  PERCY_TOKEN: ${{ secrets.PERCY_TOKEN }}
+jobs:
+  build-binaries:
```

Comment:
The GitHub Actions we've migrated previously already include a job named `compile` (in `workflows/test-core.yml`). Doesn't this job overlap significantly with that one? If there's any differences here can we merge them to that `compile` job (alternately, should `compile` be pulled out to a separate workflow?)

Reply:
It doesn't loop like `test-api` is reliant on `compile` but afaik the `compile` and `build-binaries` steps are slightly different:

1. `compile` - calls `make bootstrap && make dev`
2. `build-binaries` - compiles `ui` static assets then makes binaries

I think the biggest difference being that `build-binaries` needs that `ui` static assets steps, so I'm not sure they can be combined.
I think why `build-binaries` even exists in the first place (despite not being used downstream in the current workflow) is to fetch in the `nomad-e2e` repo [here](https://github.com/hashicorp/nomad-e2e/blob/main/circleci/scripts/fetch-binaries.sh)

Reply:
> I think the biggest difference being that build-binaries needs that ui static assets steps, so I'm not sure they can be combined.
> I think why build-binaries even exists in the first place (despite not being used downstream in the current workflow)

Hm, good point. The `compile` ends up being shorter (because compiling the UI takes so long), which is handy for unit tests. But we need the `build-binaries` with the UI to be available for launching our nightly E2E tests. So it might be worth filtering the `build-binaries` down to running just on the `main` and the `release/*` branches. 

Actually making those binaries available to the `nomad-e2e` repo's actions is a whole other question for y'all to solve :grinning: but it might be worth considering moving the `nomad-e2e` nightly runs into `nomad-enterprise` so that we can just use the binaries we build in this action for the nightly run.

Reply:
Yeah this is something we'll have to chat through when Daniel gets back - I think moving the `nomad-e2e` actions into `ent` probably makes sense since it's using those binaries anyways. I've left it for the time being with the caveat that we're moving on to the `nomad-e2e` repo next anyways, so we'll revisit this then.